### PR TITLE
Duty Scheduler Fork Related Missing Tests

### PR DESCRIPTION
**Pull request by olegshmuelov**
closes https://github.com/ssvlabs/ssv/issues/1607

also, this PR introducing a fix to a bug where the handler starts in preparation epochs range that will not trigger next period duties

### CODE REVIEW

File: operator/duties/sync_committee.go [hunk 1]
```diff
@@ -87,12 +91,8 @@ func (h *SyncCommitteeHandler) HandleDuties(ctx context.Context) {
...
 			cancel()
 
-			// If we have reached the mid-point of the epoch, fetch the duties for the next period in the next slot.
-			// This allows us to set them up at a time when the beacon node should be less busy.
-			epochsPerPeriod := h.network.Beacon.EpochsPerSyncCommitteePeriod()
-			if uint64(slot)%h.network.Beacon.SlotsPerEpoch() == h.network.Beacon.SlotsPerEpoch()/2-2 &&
-				// Update the next period if we close to an EPOCHS_PER_SYNC_COMMITTEE_PERIOD boundary.
-				uint64(epoch)%epochsPerPeriod == epochsPerPeriod-syncCommitteePreparationEpochs {
+			// if we have reached the preparation slots, prepare the next period duties in the next slot.
+			if h.remainingSLots(slot)-1 == h.preparationSlots {
```
Comment (y0sher) [inline operator/duties/sync_committee.go:95]:
```suggestion
			if h.remainingSlots(slot)-1 == h.preparationSlots {
```

File: operator/duties/sync_committee.go [hunk 2]
```diff
@@ -317,8 +317,10 @@ func calculateSubscriptions(endEpoch phase0.Epoch, duties []*eth2apiv1.SyncCommi
...
 
 func (h *SyncCommitteeHandler) shouldFetchNextPeriod(slot phase0.Slot) bool {
-	epochsPerPeriod := h.network.Beacon.EpochsPerSyncCommitteePeriod()
-	epoch := h.network.Beacon.EstimatedEpochAtSlot(slot)
-	return uint64(slot)%h.network.SlotsPerEpoch() >= h.network.SlotsPerEpoch()/2-1 &&
-		uint64(epoch)%epochsPerPeriod >= epochsPerPeriod-syncCommitteePreparationEpochs
+	return h.remainingSLots(slot) <= h.preparationSlots
+}
+
+func (h *SyncCommitteeHandler) remainingSLots(slot phase0.Slot) uint64 {
```
Comment (y0sher) [inline operator/duties/sync_committee.go:323]:
```suggestion
func (h *SyncCommitteeHandler) remainingSlots(slot phase0.Slot) uint64 {
```

File: operator/duties/sync_committee.go (hunk 1 above)
Comment (olegshmuelov) [reply operator/duties/sync_committee.go]: ?

**Review by y0sher — APPROVED** [review]

**Review by moshe-blox — APPROVED** [review]
very thorough and readable tests, well done 💪

**Pull request merged**